packages/block-editor/src/components/provider/index.js

```diff
@@ -24,7 +25,7 @@ function BlockEditorProvider( props ) {
 	// Syncs the entity provider with changes in the block-editor store.
 	useBlockSync( props );
 
-	return children;
+	return <BlockRefsProvider>{ children }</BlockRefsProvider>;
```

Comment:
Since this is now a Provider inside the state provider, it makes me wonder whether we really need this new provider and whether we can just use the store to keep track of these refs.

Reply:
Hm, I'm not sure. Then we'd have refs and elements in the redux state?

Reply:
Is that an issue? 🤷‍♂️That doesn't bother me personally,  it's not a part of the state that is meant to be serializable

Reply:
To clarify, it doesn't bother me as right now, I'm just wondering whether it's just the same thing and we could avoid the extra state/provider

Reply:
I can try, but we'd still keep the mutable maps because otherwise we're rerendering the editor too much on load